### DISCUSSION TITLE
How to solve the maximization of the dual problem?

### DISCUSSION THREAD

**Discussion by shiquan815** [casadi/casadi#3138] [Q&A] [answered]
The standard cost function provided by casadi is to solve the minimization problem. How to solve the maximization of the dual problem? Is there a related demo?

**Comment by jaeandersson** [comment, marked as answer]
You would need to formulate the dual counterpart yourself before sending it to a solver. That's possible for some problem classes, like convex QPs, but not others

**Reply by shiquan815** [reply]
I'm sorry, I'm a little confused. In casadi, is there a function that directly solves the dual problem?
For example: I want to solve the distance between polyhedrons, min h= (z_i-z_j) ^2, s.t. A_i*z_i<=b_i & A_j*z_j<=b_j, and its dual problem can be formulated as: max L=-1/4 lamda_i* A_i* A_i*lamda_i-lamda_i*b_i-lamda_j*b_j;  s.t. lamda_i*A_i+lamda_j*A_j=0,lamda_i, lamda_j>=0, how to find this maximum using CasAdi?

**Reply by jaeandersson** [reply]
CasADi has no explicit support for formulating the dual problem. If you can do the reformulation yourself, and the result is a standard problem class, you can solve it with the normal solvers. In your case above, it looks like the dual is just a QP, so you can solve it with a standard QP solver. To solve a maximization problem for $f(x)$, just minimize $-f(x)$ instead.

**Reply by shiquan815** [reply]
THANK YOU VERY MUCH!
